src/Nethereum.Util/TaskExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Nethereum.Util
+{
+    /// <summary>
+    /// https://stackoverflow.com/a/15136833/4993930
+    /// </summary>
+    public static class TaskExtensions
+    {
+        /// <summary>
+        /// Enumerates a collection in parallel and calls an async method on each item. Useful for making 
+        /// parallel async calls, e.g. independent web requests when the degree of parallelism needs to be
+        /// limited.
+        /// </summary>
+        public static Task ForEachAsync<T>(this IEnumerable<T> source, int degreeOfParalellism, Func<T, Task> action)
+        {
+            return Task.WhenAll(Partitioner.Create(source).GetPartitions(degreeOfParalellism).Select(partition => Task.Run(async () =>
+            {
+                using (partition)
+                    while (partition.MoveNext())
+                        await action(partition.Current);
+            })));
+        }
+        public static Task ForEachAsync<T>(this IEnumerable<T> source, Func<T, Task> action)
+        {
+            return Task.WhenAll(Partitioner.Create(source).GetPartitions(Environment.ProcessorCount).Select(partition => Task.Run(async () =>
```

Comment:
The processing code is normally I/O bound, not CPU bound. So ProcessorCount should not be relevant here.

Reply:
@sergey-s-s it could be put a strategy in place for the parallel processing, that was the original thought, so it does not need to be parallel if not wanted.

Reply:
@sergey-s-s  https://github.com/Nethereum/Nethereum/commit/6699ae6309f27232c4aba7bf158416546233ab28